### DISCUSSION TITLE
[HELP] Issue while Using it as a submodule.

### DISCUSSION THREAD

**Discussion by Shabinder** [JakeWharton/mosaic#34] [Q&A]
Earlier I copy pasted , dependencies.gradle , **mosaic/mosaic** folder and substituted modules with projects , and all worked fine

now, I wanted to escape from this manual work , So tried to add it as a 
`git add submodule  https://github.com/JakeWharton/mosaic`
and It worked and added mosaic project as a submodule , so I did the following in my project's settings.gradle.kts

```
includeBuild("mosaic/mosaic") {
    dependencySubstitution {
        substitute(module("com.jakewharton.mosaic:mosaic-gradle-plugin")).with(project(":mosaic-gradle-plugin"))
        substitute(module("com.jakewharton.mosaic:mosaic-runtime")).with(project(":mosaic-runtime"))
        substitute(module("com.jakewharton.mosaic:compose-compiler")).with(project(":compose:compiler"))
    }
}
```
and added the **plugin** as following: 
```
plugins {
    // other plugins
    id("com.jakewharton.mosaic")
}
```
then did a **gradle sync successfully** and app **run** (used your sample counter app's code) also **successfully** :smile: 

 BUT IDE doesnt recognise code reference , however builds successfully, any help would be appreciated , If I can provide more info , I would be happy to do that...

![image](https://user-images.githubusercontent.com/44765050/123472742-2fd3d080-d615-11eb-8187-9f384dab5748.png)

**Comment by Shabinder** [comment]
IDE is only recognising following : 
![image](https://user-images.githubusercontent.com/44765050/123472915-745f6c00-d615-11eb-9c5b-0fb790319e2c.png)

**Comment by JakeWharton** [comment]
I don't really have an answer for you. I also sometimes have problems with the IDE and the samples even locally which uses the same setup. Sometimes it fixes itself, sometimes it doesn't and I'm stuck in a glorified text editor. Would love a proper fix, but I suspect it's inside the Kotlin plugin / IntelliJ IDEA itself and nothing specific to this project.
